packages/core/src/data-centralization/dataCentralizationRuntime.ts

```diff
@@ -96,55 +96,73 @@ export async function dataCentralizationRuntime({
+	const closeChallengeListener = listenForChallenges(
+		async (challengeNumber: bigint, challenge: Challenge, event?: any) => {
+			try {
+				const startTime = new Date().getTime();
+				const PoolModel = mongoose.models.Pool || mongoose.model<IPool>('Pool', PoolSchema);
 
-		const slackStartMessage = `Starting pool sync update for challenge ${challengeNumber}`;
-		sendSlackNotification(slackWebHookUrl, slackStartMessage, logFunction);
+				const slackStartMessage = `Starting pool sync update for challenge ${challengeNumber}`;
+				await sendSlackNotification(slackWebHookUrl, slackStartMessage, logFunction);
 
-		const graphUpdateStartTime = new Date().getTime();
+				const graphUpdateStartTime = new Date().getTime();
 
-		const updatedPools = await retry(() => getRewardRatesFromGraph([]));
+				const updatedPools = await retry(() => getRewardRatesFromGraph([]));
 
-		const graphUpdateEndTime = new Date().getTime();
+				const graphUpdateEndTime = new Date().getTime();
 
-		const mongoInsertStartTime = new Date().getTime();
+				const mongoInsertStartTime = new Date().getTime();
 
-		for (const updatedPool of updatedPools) {
-			const checksumAddress = getAddress(updatedPool.poolAddress);
+				for (const updatedPool of updatedPools) {
+					const checksumAddress = getAddress(updatedPool.poolAddress);
 
-			await PoolModel.updateOne(
-				{ poolAddress: checksumAddress },
-				{
-					$set: {
-						esXaiRewardRate: updatedPool.averageDailyEsXaiReward,
-						keyRewardRate: updatedPool.averageDailyKeyReward,
-						totalEsXaiClaimed: updatedPool.totalEsXaiClaimed
-					}
-				},
-			);
+					await PoolModel.updateOne(
+						{ poolAddress: checksumAddress },
+						{
+							$set: {
+								esXaiRewardRate: updatedPool.averageDailyEsXaiReward,
+								keyRewardRate: updatedPool.averageDailyKeyReward,
+								totalEsXaiClaimed: updatedPool.totalEsXaiClaimed
+							}
+						},
+					);
+				}
+				const mongoInsertEndTime = new Date().getTime();
+				const totalSeconds = mongoInsertEndTime - startTime;
+				const totalGraphSeconds = graphUpdateEndTime - graphUpdateStartTime;
+				const totalMongoSeconds = mongoInsertEndTime - mongoInsertStartTime;
+				const slackMessage = `Finished pool sync update for ${updatedPools.length} pools in ${totalSeconds}ms. Graph update took ${totalGraphSeconds}ms. Mongo insert took ${totalMongoSeconds}ms.`;
+
+				await sendSlackNotification(slackWebHookUrl, slackMessage, logFunction);
+			} catch (error) {
+				const errorMessage = error instanceof Error ? error.message : String(error);
+				await sendSlackNotification(slackWebHookUrl, `Error in challenge listener: ${errorMessage}`, logFunction);
+				logFunction(`Error in challenge listener: ${errorMessage}`);
+			}
+		},
+		async (error: Error) => {
+			const errorMessage = `Error in listenForChallenges: ${error.message}`;
+			await sendSlackNotification(slackWebHookUrl, errorMessage, logFunction);
```

Comment:
Please revise the slack notifications, error / restarts should be send with the `@channel`, I think you have to do that by adding `<!channel>>` in front of the message text.

Reply:
This has been added, thank you for the guidance.